csrc/tensor_view.cpp

```diff
@@ -1023,7 +1023,12 @@ TensorView* TensorView::rFactor(const std::vector<int>& axes) {
         this_mma->inA(),
         this_mma->inB(),
         this_mma->init(),
-        this_mma->options());
+        this_mma->options(),
+        this_mma->mAxes(),
+        this_mma->nAxes(),
+        this_mma->kAxes(),
```

Comment:
Hmm, I don't think these axes remain the same. When you do rFactor, the dimensionality of the tensor is changed, and some part of the `k`s becomes `batch`. I think you should not copy these axes here. You should do a new computation to re-infer these axes from the new inputs and outputs.

Reply:
@zasdfgbnm I updated code as I mentioned above (attributes are always initialized, we only provide `options` in the second constructor) but some tests are failing now due to missing `layout` attribute initialized in `MmaOp` - this only happens in the lowering phase where `Val`s are `TensorIndex` type.

You mentioned earlier, some `K`s may move to `batches` axes due to modifications done to tensors but I think that the core input layouts shouldn't change. How about propagating the inputs layout along `options` in the `MmaOp` constructor used in codegen?

Reply:
I think this makes sense.